src/isoStringScalar/isoStringScalar.ts

```diff
@@ -0,0 +1,86 @@
+import { GraphQLScalarType, Kind } from 'graphql';
+import { isValid, parseISO, toDate } from 'date-fns';
+import {
+  InternalServerError,
+  UserInputError,
+} from '../errorHandler/errorHandler';
+
+export const isoStringScalar = new GraphQLScalarType({
+  name: 'ISOString',
+  description: 'ISOString custom scalar type',
+
+  /**
+   * Converts server-side TS Date representation to a JSON-compatible, ISO-8601-compliant
+   * UTC Datetime String for Apollo Server to include in an operation response.
+   *
+   * @param value - TS Date Object as generated by Data Store Client & validated by mysqlDateConvert.
+   * @returns ISO-8601-compliant, UTC-based Datetime String.
+   */
+  serialize(value: Date | null): string | null {
+    if (value === null) {
+      return null;
+    }
+
+    if (value instanceof Date && isNaN(value.valueOf())) {
+      throw new InternalServerError(
```

Comment:
I know this is exactly what I suggested when we talked about this before, but I'm going to ask a question here as well that hopefully other reviewers can answer.

It's my understanding that the `0000-00-00 00:00:00.000Z` (or whatever the exact string is) values in the database are a stand in for a `None` value (a string value that we understand as un-initialized or maybe even equivalent to falsey even though v8 doesn't see it this way).  Do we want to throw an error here, or could we be okay with returning null? I'm a little worried about this being something that blocks successful responses on those bulk APIs, especially since it's after the db transaction has been concluded, so it's not like we can just roll it back.

Reply:
There are times when a null date is valid. For example archivedAt (`time_read` in db) is null/special 0 time string if the save is not archived. So I think we should allow null values to pass through without error.

Reply:
Cool. I'm on board with that. A comment where it's returning null for that NaN case would be great and wrap this up nicely.

Reply:
So that's been my understanding and proposal the whole time - 0000-00-00 values in the database mean None in the server semantically - e.g., a date hasn't been set for some reason. So its not an error, its a null.

Reply:
Added note.